src/performance_test_fixture.cpp

```diff
@@ -102,6 +106,19 @@ void PerformanceTest::on_realloc(
 void PerformanceTest::reset_heap_counters()
 {
   allocation_count = 0;
+  are_performance_measurements_active = true;
+}
+
+void PerformanceTest::pause_performance_measurements(::benchmark::State & state)
+{
+  state.PauseTiming();
+  are_performance_measurements_active = false;
+}
+
+void PerformanceTest::resume_performance_measurements(::benchmark::State & state)
+{
+  are_performance_measurements_active = true;
+  state.ResumeTiming();
 }
```

Comment:
What would you think about removing the state-specific parts of this function and instead defining a macro or an inline function that does this to avoid the extra function call overhead?

Reply:
Another option would be to make this even "prettier" with a sort of scoping macro:
```c++
#define DO_NOT_TRACK(state, code) \
  do { \
    state.PauseTiming(); \
    pause_performance_measurements(); \
    { code } \
    resume_performance_measurements(); \
    state.ResumeTiming(); \
  } while (0);

...

for (auto _ : state) {
  do_the_thing();
  DO_NOT_TRACK(state, {
    reset_the_thing();
  });
}
```

Reply:
I like the idea of the macro, since it also mirrors what I've done with the fault injection where we needed to something similar. I spent way to much time writing benchmarks to prove to myself that setting the boolean in a member function does not impact measurements. Probably because compiler often inlines small functions anyway.

Nonetheless, I have now added benchmarks to measure Pause/Resume timing and the new macro `PERFORMANCE_TEST_FIXTURE_PAUSE_MEASUREMENTS` based on your example. I've rebased this PR onto #11.